x-pack/plugins/cases/public/components/all_cases/columns.tsx

```diff
@@ -51,35 +53,106 @@ const TagWrapper = styled(EuiBadgeGroup)`
 const renderStringField = (field: string, dataTestSubj: string) =>
   field != null ? <span data-test-subj={dataTestSubj}>{field}</span> : getEmptyTagValue();
 
-interface GetCasesColumn {
-  actions: Array<DefaultItemIconButtonAction<Case>>;
-  caseDetailsNavigation: CasesNavigation<CaseDetailsHrefSchema, 'configurable'>;
+export interface GetCasesColumn {
+  caseDetailsNavigation?: CasesNavigation<CaseDetailsHrefSchema, 'configurable'>;
   filterStatus: string;
-  isModal: boolean;
+  handleIsLoading: (a: boolean) => void;
+  refreshCases?: (a?: boolean) => void;
+  showActions: boolean;
 }
-export const getCasesColumns = ({
-  actions,
+export const useCasesColumns = ({
   caseDetailsNavigation,
   filterStatus,
-  isModal,
+  handleIsLoading,
+  refreshCases,
+  showActions,
 }: GetCasesColumn): CasesColumns[] => {
-  const columns = [
+  const { loading: isLoadingCases, dispatchUpdateCaseProperty } = useGetCases();
+  // Delete case
+  const {
+    dispatchResetIsDeleted,
+    handleOnDeleteConfirm,
+    handleToggleModal,
+    isDeleted,
+    isDisplayConfirmDeleteModal,
+    isLoading: isDeleting,
+  } = useDeleteCases();
+
+  const [deleteThisCase, setDeleteThisCase] = useState<DeleteCase>({
+    id: '',
+    title: '',
+    type: null,
+  });
+  const confirmDeleteModal = useMemo(
+    () => (
+      <ConfirmDeleteCaseModal
+        caseTitle={deleteThisCase.title}
+        isModalVisible={isDisplayConfirmDeleteModal}
+        isPlural={false}
+        onCancel={handleToggleModal}
+        onConfirm={handleOnDeleteConfirm.bind(null, [deleteThisCase])}
+      />
+    ),
+    [deleteThisCase, isDisplayConfirmDeleteModal, handleToggleModal, handleOnDeleteConfirm]
+  );
+  const toggleDeleteModal = useCallback(
+    (deleteCase: Case) => {
+      handleToggleModal();
+      setDeleteThisCase({ id: deleteCase.id, title: deleteCase.title, type: deleteCase.type });
+    },
+    [handleToggleModal]
+  );
+
+  const handleDispatchUpdate = useCallback(
+    (args: Omit<UpdateCase, 'refetchCasesStatus'>) => {
+      dispatchUpdateCaseProperty({
+        ...args,
+        refetchCasesStatus: () => {
+          if (refreshCases != null) refreshCases();
+        },
+      });
+    },
+    [dispatchUpdateCaseProperty, refreshCases]
+  );
+
+  const actions = useMemo(
```

Comment:
I don't think this needs a useMemo

Reply:
why not? otherwise it is called each render

Reply:
You can ignore this. I dug into the `actions` call a bit deeper. What I was thinking was that since all it's doing is just setting properties and returning an object, the actual call isn't that computationally expensive, but I missed maintaining the referential equality when it's being passed to the `CasesTable` and `expandedRowMap`.